extensions/amp-a4a/0.1/template-validator.js

```diff
@@ -101,4 +94,52 @@ export class TemplateValidator extends Validator {
               }));
         });
   }
+
+  /**
+   * Adds the mustache or analytics extensions if not already installed.
+   * TODO(alabiaga): If not installed we load the 0.1 versions of the
+   * extensions. Should we default to the latest version?
+   * @param {?../../../extensions/amp-a4a/0.1/amp-ad-type-defs.CreativeMetaDataDef} creativeMetadata
+   * @param {!./amp-ad-type-defs.AmpTemplateCreativeDef} parsedResponseBody
+   */
+  addAnalyticsOrMustacheIfApplicable(creativeMetadata, parsedResponseBody) {
+    if (parsedResponseBody.analytics) {
+      this.addExtensionIfApplicable_(
+          creativeMetadata,
+          'amp-analytics',
+          urls.cdn + '/v0/amp-analytics-0.1.js');
+    }
+    this.addExtensionIfApplicable_(
+        creativeMetadata,
+        'amp-mustache',
+        urls.cdn + '/v0/amp-mustache-0.1.js');
+  }
+
+  /**
+   * @param {?../../../extensions/amp-a4a/0.1/amp-ad-type-defs.CreativeMetaDataDef} creativeMetadata
+   * @param {string} extensionId
+   * @param {string} extensionSrc
+   */
+  addExtensionIfApplicable_(creativeMetadata, extensionId, extensionSrc) {
+    if (creativeMetadata) {
+      let ext = [];
+      const {extensions, customElementExtensions} = creativeMetadata;
+      if (extensions && extensions.length) {
+        ext = extensions;
+      } else if (customElementExtensions && customElementExtensions.length) {
+        ext = customElementExtensions;
+      }
+      if (ext.indexOf(extensionId) == -1) {
```

Comment:
I don't believe this can work in the `extensions` case, they're objects not strings.